plugin-server/src/cdp/consumers/cdp-events.consumer.ts

```diff
@@ -121,27 +127,38 @@ export class CdpEventsConsumer extends CdpConsumerBase {
             const validInvocations: CyclotronJobInvocationHogFunction[] = []
 
             // Iterate over adding them to the list and updating their priority
-            possibleInvocations.forEach((item) => {
+            for (const item of possibleInvocations) {
                 // Disable invocations for teams that don't have the addon (for now just metric them out..)
 
-                if (
-                    !teamsById[`${item.teamId}`]?.available_features.includes('data_pipelines') &&
-                    item.hogFunction.is_addon_required
-                ) {
-                    // NOTE: This counter is temporary until we are confident in enabling this
-                    counterMissingAddon.labels({ team_id: item.teamId }).inc()
-                    // TODO: Later add the below code
+                const isQuotaLimited = await this.hub.quotaLimiting.isTeamQuotaLimited(item.teamId, 'cdp_invocations')
+
+                // The legacy addon was not usage based so we skip dropping if they are on it
+                const isTeamOnLegacyAddon = !!teamsById[`${item.teamId}`]?.available_features.includes('data_pipelines')
```

Comment:
Not 100% sure if the underlying feature will change so we might need a different way of detecting legacy add-on vs new usage based product